### PR TITLE
CODENVY-554 if factory has an ID, no need to give again parameters as…

### PR DESCRIPTION
**Pull request by benoitf**
… factory is already resolved

Change-Id: I12bc304b07ddb0320c76bf69c49c18ae7aba4555
Signed-off-by: Florent BENOIT <fbenoit@codenvy.com>

### CODE REVIEW

File: assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java [hunk 1]
```diff
@@ -84,23 +84,28 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
...
-                    if (firstParam) {
-                        redirectUrl = redirectUrl.concat("?");
-                        firstParam = false;
-                    } else {
-                        redirectUrl = redirectUrl.concat("&");
+        // factory has id, no need to use parameters
+        if (requestedFactory.getId() != null) {
+            redirectUrl += "?id=" + requestedFactory.getId();
+        } else {
+            boolean firstParam = true;
```
Comment (skabashnyuk) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: why don't you add whole query string to redirect url?

Comment (benoitf) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: ?

Comment (skabashnyuk) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: something like redirectUrl= redirectUrl +"?"+req.getQueryString()

**Comment by skabashnyuk** [comment]
ок

File: assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java (hunk 1 above)
Comment (benoitf) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: fixed, thanks

**Pull request merged**